### PR TITLE
swagger-jsdoc: use true typescript import statements instead of require

### PR DESCRIPTION
**Pull request by iliuta**
This module requires use of require('') style imports, which is not compliant to ES6 style imports.
The pull request fixes this by using TS export statement instead of export= .

Documentation and test has been adapted accordingly.

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [ ] Provide a URL to documentation or source code which provides context for the suggested changes: https://palantir.github.io/tslint/rules/no-require-imports/
- [ ] Increase the version number in the header if appropriate.
- [ ] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

**Review by drGrove — APPROVED** [review]

**Review by ghost — CHANGES_REQUESTED** [review]

File: types/swagger-jsdoc/index.d.ts [hunk 1]
```diff
@@ -6,8 +6,8 @@
 
 /* =================== USAGE ===================
 
-    import * as express from "express"
-    import swaggerJSDoc = require('swagger-jsdoc');
+    import * as express from 'express';
+    import { swaggerJSDoc } from 'swagger-jsdoc';
```
[the remark this answers is not in the document]
Comment (iliuta) [reply types/swagger-jsdoc/index.d.ts]: Hi Andy, 

Thanks for your reply. However, I'm not sure I understand.

I've just tested:
```
import { swaggerJSDoc } from 'swagger-jsdoc';
let options = {};
let swaggerSpec = swaggerJSDoc.swaggerJSDoc(options);
```

is valid and compiles into 

```
const swagger_jsdoc_1 = require("swagger-jsdoc");
let swaggerSpec = swagger_jsdoc_1.swaggerJSDoc(options)
```
On the other hand,
```
import  swaggerJSDoc = require('swagger-jsdoc');
let options={};
let swaggerSpec = swaggerJSDoc(options);
```
is invalid and requires change in order to be valid:
```
let swaggerSpec = swaggerJSDoc.swaggerJSDoc(options);
```

[the remark this answers is not in the document]
Comment (iliuta) [reply types/swagger-jsdoc/index.d.ts]: Hi, thanks for your reply. I first replied, but with further analysis I prefered to delete.

I changed the definition file in order to allow
```
import * as swaggerJSDoc from 'swagger-jsdoc'
```
Would that be more suitable?

[the remark this answers is not in the document]
Comment (andy-ms) [reply types/swagger-jsdoc/index.d.ts]: I don't think we're encouraging that `namespace {}` hack anymore (@DanielRosenwasser right?) -- why do you want to use `import *` and not `import =`? If you're in an es6-like environment, maybe `--allowSyntheticDefaultImports` or `--esModuleInterop` would be more appropriate?

[the remark this answers is not in the document]
Comment (DanielRosenwasser) [reply types/swagger-jsdoc/index.d.ts]: Correct, no more `namespace foo {}` hacks.

**Comment by DanielRosenwasser** [comment]
Just as a heads up, your commits don't seem to be associated with your GitHub account. While this isn't technically a problem, you might care if you want more appropriate attribution. You can either make sure your GitHub account is associated with the email address you're using for your commits, or rebase and amend your commits to fix the author name and email.

If you don't care about any of this stuff, I can always just pull in the changes.

**Comment by minestarks** [comment]
@iliuta Could you update with the requested revisions re: the module import? Thanks.

**Comment by mhegazy** [comment]
Thanks for your contribution. This PR has unaddressed comments and can not be merged in at the time being. For housekeeping purposes we are closing stale PRs. If you'd still like to merge this code in, please open a new PR that has been merged and rebased with the `master` branch.

**Pull request closed without merging**